### PR TITLE
fix avatar flash

### PR DESCRIPTION
**Pull request by michaelkirk**
Fixed the avatar flicker in the inbox and added debug actions to create a bunch of contact threads.

### CODE REVIEW

File: Signal/src/ViewControllers/InboxTableViewCell.m [hunk 1]
```diff
@@ -227,16 +227,8 @@ - (void)configureWithThread:(TSThread *)thread
...
-    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        UIImage *avatar =
-            [OWSAvatarBuilder buildImageForThread:thread contactsManager:contactsManager diameter:kavatarViewDiameter];
-        dispatch_async(dispatch_get_main_queue(), ^{
-            if ([_threadId isEqualToString:threadIdCopy]) {
-                self.avatarView.image = avatar;
-            }
-        });
-    });
+    self.avatarView.image =
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/InboxTableViewCell.m]: We initially talked about having this being async in the event of a cache miss, but in my testing, after fixing the caching, even on my iphone5 with 1k threads, this scrolls smoothly without egregious CPU (4% in generating avatars - when scrolling wildly, but it trends to zero when you are scrolling back and forth due to the cache hits).

I decided it wasn't worth the complexity.

File: Signal/src/Models/OWSContactAvatarBuilder.m [hunk 2]
```diff
@@ -95,7 +95,7 @@ - (UIImage *)buildDefaultImage
                                                                        textColor:[UIColor whiteColor]
                                                                             font:[UIFont ows_boldFontWithSize:fontSize]
                                                                         diameter:self.diameter] avatarImage];
-    [self.contactsManager.avatarCache setObject:image forKey:self.signalId];
+    [self.contactsManager.avatarCache setObject:image forKey:cacheKey];
```
Comment (michaelkirk) [inline Signal/src/Models/OWSContactAvatarBuilder.m]: We were effectively never caching generated avatars =(

Comment (charlesmchen) [inline Signal/src/Models/OWSContactAvatarBuilder.m]: =(

File: Signal/src/ViewControllers/InboxTableViewCell.m (hunk 1 above)
Comment (charlesmchen) [inline Signal/src/ViewControllers/InboxTableViewCell.m]: Okay.  Please make sure that perf testing is part of the test plan.

Comment (charlesmchen) [inline Signal/src/ViewControllers/InboxTableViewCell.m]: Specifically, I'm assuming that perf is a function of input avatar size, so maybe we should modify the "create fake contacts" logic to include a randomly generated image, whose file size/content sizes vary.

**Pull request merged**